nemo/collections/asr/metrics/wer.py

```diff
@@ -52,88 +49,105 @@ def word_error_rate(hypotheses: List[str], references: List[str], use_cer=False)
             h_list = h.split()
             r_list = r.split()
         words += len(r_list)
-        scores += __levenshtein(h_list, r_list)
+        scores += editdistance.eval(h_list, r_list)
     if words != 0:
         wer = 1.0 * scores / words
     else:
         wer = float('inf')
     return wer
 
 
-def __ctc_decoder_predictions_tensor(tensor, labels):
-    """
-    Decodes a sequence of labels to words
-    """
-    blank_id = len(labels)
-    hypotheses = []
-    labels_map = dict([(i, labels[i]) for i in range(len(labels))])
-    prediction_cpu_tensor = tensor.long().cpu()
-    # iterate over batch
-    for ind in range(prediction_cpu_tensor.shape[0]):
-        prediction = prediction_cpu_tensor[ind].detach().numpy().tolist()
-        # CTC decoding procedure
-        decoded_prediction = []
-        previous = len(labels)  # id of a blank symbol
-        for p in prediction:
-            if (p != previous or previous == blank_id) and p != blank_id:
-                decoded_prediction.append(p)
-            previous = p
-        hypothesis = ''.join([labels_map[c] for c in decoded_prediction])
-        hypotheses.append(hypothesis)
-    return hypotheses
-
-
-def monitor_asr_train_progress(tensors: list, labels: list, eval_metric='WER', ctc_decode=True):
+class AverageTextWER(TensorMetric):
     """
-    Takes output of greedy ctc decoder and performs ctc decoding algorithm to
-    remove duplicates and special symbol. Prints sample to screen, computes
-    and logs AVG WER to console
+    This metric computes nominator and denominator for Average Text Word Error Rate.
+    When doing distributed training/evaluation the result of res=AverageTextWER(predictions, targets, target_lengths) calls
+    will be all-reduced between all workers using SUM operations.
+    Here contains two numbers res=[wer_nominator, wer_denominator]
+
+    If used with PytorchLightning LightningModule, include wer_nominator and wer_denominator inside validation_step results.
+    Then aggregate (sum) then at the end of validation epoch to correctly compute validation WER.
```

Comment:
Should we also make it explicit in the docstring that to get WER one would need to do `WER=res[0]/res[1]` after summing both?